### DISCUSSION TITLE
Converting Spectre file to Seurat Object after running FlowSOM

### DISCUSSION THREAD

**Discussion by pavitraviswanath** [ImmuneDynamics/Spectre#110] [Help] [answered]
Hi, 

Thank you again for a great package! I was wondering if there is a way to convert the Spectre data.table file obtained after running FlowSOM (or maybe even after t-sne or umap analysis) into a Seurat object? The main reason I'd like to do this is to use the mouse cell atlas as a reference for integration and label transfer with Seurat.

Thanks!

**Comment by tomashhurst** [comment]
@pavitraviswanath yes absolutely you can! We are doing a bunch of integration of cytometry and single-cell multiomic data with some new functions and workflows that will come out with v2 (it works with v1 now, but it's just more fiddly). If you are interested in trying it out, we could certainly give you early access! In the meantime, creating a Seurat object from the data.table is pretty straightforward. I'll put some instructions online and send you a link -- I've tested a few of the integration approaches after doing this conversion and they seem to work just fine.

**Reply by pavitraviswanath** [reply]
Thank you so much Tom! I would love to try it out if I do end up getting early access. :)

**Reply by denvercal1234GitHub** [reply]
@tomashhurst @pavitraviswanath -- I am also very interested since we have spectral flow data and would love to be able to integrate it to our and others' scRNAseq data. Would you mind letting me know also whether the v2 with these features is available? Thank you!

**Comment by tomashhurst** [comment, marked as answer]
@pavitraviswanath 

To get you started:

```
library('Spectre')
library('Seurat')
```

Here's a quick example with the demo dataset:

```
Spectre::demo.clustered
as.matrix(names(Spectre::demo.clustered))
```

Here you pick whichever columns you want for the primary 'data' (i.e, in this case replacing counts)

```
use.cols <- names(Spectre::demo.clustered)[c(11:19)]
use.cols
```

```
dat <- Spectre::demo.clustered[,use.cols, with = FALSE]
dat
```

Here we can turn the data into a matrix,  then into a sparse matrix, and transpose it.

```
dat <- as.matrix(dat)
dat <- Matrix::Matrix(dat, sparse = TRUE)
rownames(dat) <- paste0('Cell', '-', c(1:nrow(dat)))
dat <- Matrix::t(dat)
dat
```

We can then use this input to initialise a Seurat object.

```
srt <- CreateSeuratObject(counts = dat, assay = 'cyto')
srt
```

You can then explore the object and add extra data

```
srt@assays$cyto
```
